src/whylogs/core/columnprofile.py

```diff
@@ -112,6 +112,9 @@ def track(self, value, character_list=None, token_method=None):
         if isinstance(value, str):
             self.string_tracker.update(value, character_list=character_list, token_method=token_method)
         # TODO: Implement real typed data conversion
+
+        self.constraints.update(value)
```

Comment:
For example, we have schema tracker count the types, so wanted to see if we change the "inferred data type" for example.

Reply:
This seems like it does, I think @MilenaTrajanoska changed this for the problem with the yaml.safe_load function. She should confirm though.

Reply:
It would not change the inferred data type metric, but there might be an inconsistency. For example if we have '1.2' as a string it would check it in a string matching pattern, but later on it will be converted to float 1.2 which would update the number trackers instead of the string trackers. We could instead check the type of the value and if it's a string, see if any of the constraints are MATCH or NON_MATCH, then skip the conversion and pass it as a string to the trackers, and if the column does not have string based constraints, we would do type conversion  first and pass it to the constraints and trackers as the inferred type.

Reply:
Ok, thanks for clarifying. I'm trying to make sure we preserve the existing behavior in the trackers. We can change things for the constraints, but changing behavior in the trackers would be a disruptive change. If it is easier we can have two methods rather then the one "update" on `self.constraints` and introduce an update_typed() so that we can update some constraints with the pre-conversion raw string (and the string matching/regex constraints would go there), and then the other updates can occur in the normal location post type conversion.